ansible_runner/runner_config.py

```diff
@@ -192,13 +202,38 @@ def prepare(self):
         self.env['ANSIBLE_HOST_KEY_CHECKING'] = 'False'
         self.env['AWX_ISOLATED_DATA_DIR'] = self.artifact_dir
 
+        if self.resource_profiling:
+            callback_whitelist = os.environ.get('ANSIBLE_CALLBACK_WHITELIST', '').strip()
+            self.env['ANSIBLE_CALLBACK_WHITELIST'] = ','.join(filter(None, [callback_whitelist, 'cgroup_perf_recap']))
+            self.env['CGROUP_CONTROL_GROUP'] = '{}/{}'.format(self.resource_profiling_base_cgroup, self.ident)
+            if self.resource_profiling_results_dir:
+                cgroup_output_dir = self.resource_profiling_results_dir
+            else:
+                cgroup_output_dir = os.path.normpath(os.path.join(self.private_data_dir, 'profiling_data'))
+
+            # Create results directory if it does not exist
+            if not os.path.isdir(cgroup_output_dir):
+                os.mkdir(cgroup_output_dir, stat.S_IREAD | stat.S_IWRITE | stat.S_IEXEC)
+
+            self.env['CGROUP_OUTPUT_DIR'] = cgroup_output_dir
+            self.env['CGROUP_OUTPUT_FORMAT'] = 'json'
+            self.env['CGROUP_CPU_POLL_INTERVAL'] = str(self.resource_profiling_cpu_poll_interval)
+            self.env['CGROUP_MEMORY_POLL_INTERVAL'] = str(self.resource_profiling_memory_poll_interval)
+            self.env['CGROUP_PID_POLL_INTERVAL'] = str(self.resource_profiling_pid_poll_interval)
+            self.env['CGROUP_FILE_PER_TASK'] = 'True'
+            self.env['CGROUP_WRITE_FILES'] = 'True'
+            self.env['CGROUP_DISPLAY_RECAP'] = 'False'
+
         self.env['PYTHONPATH'] = python_path + callback_dir
         if self.roles_path:
             self.env['ANSIBLE_ROLES_PATH'] = ':'.join(self.roles_path)
 
         if self.process_isolation:
             self.command = self.wrap_args_with_process_isolation(self.command)
 
+        if self.resource_profiling and self.execution_mode == ExecutionMode.ANSIBLE_PLAYBOOK:
```

Comment:
Sorry, I didn't notice this from my previous reviews but why is this limited to just `ANSIBLE_PLAYBOOK`? Seems like it would be useful for ad-hoc mode also?  If not then maybe we should generate an error rather than silently not attaching the cgroup details.

Reply:
@matburt - 👍 Filed:
https://github.com/ansible/ansible-runner/issues/356